tools/filter.py

```diff
@@ -0,0 +1,150 @@
+        """
+        Adds a filter from a string
+
+        Args:
+            filter_string (string): filter provided as command:pattern
+        """
+        _filter_split = filter_string.split(':')
+        if len(_filter_split) != 2:
+            log(f"filter string '{filter_string}' does not conform to format 'component:pattern'")
+            raise EESSIBotActionFilterError(f"filter '{filter_string}' does not conform to format 'component:pattern'")
+        if len(_filter_split[1]) == 0:
+            log(f"pattern in filter string '{filter_string}' is empty")
+            raise EESSIBotActionFilterError(f"pattern in filter string '{filter_string}' is empty")
+        self.add_filter(_filter_split[0], _filter_split[1])
+
+    def remove_filter(self, component, pattern):
+        """
+        Removes all elements matching the filter given by (component, pattern)
+
+        Args:
+            component (string): any prefix of 'architecture', 'instance', 'job' or 'repository'
+            pattern (string): regex that is applied to a string representing the component
+        """
+        index = 0
+        for _filter in self.action_filters:
+            if _filter.component.startswith(component) and pattern == _filter.pattern:
+                log(f"removing filter ({_filter.component}, {pattern})")
+                self.action_filters.pop(index)
+            else:
+                index += 1
+
+    def to_string(self):
+        """
+        Convert filters to string
+        """
+        filter_str_list = []
+        for _filter in self.action_filters:
+            cm = _filter.component
+            re = _filter.pattern
+            filter_str_list.append(f"{cm}:{re}")
+        return " ".join(filter_str_list)
+
+    def check_filters(self, context):
+        """
+        Checks filters for a given context which is defined by one to four
+        components (architecture, instance, job, repository)
+
+        Args:
+            context (dict) : dictionary that maps component to value
+
+        Returns:
+            True if all defined filters match corresponding component in given
+            context
+        """
+        # if no filters are defined we return True
+        if len(self.action_filters) == 0:
+            return True
+
+        # we have at least one filter which has to match or we return False
+        check = False
+
+        # examples:
+        #   arch:intel instance:AWS --> rebuild for all repos for intel architectures on AWS
+        #   arch:generic --> rebuild for all repos for generic architectures anywhere
+        #   repository:nessi.no-2022.11 --> disable all builds for nessi.no-2022.11
+        # we iterate over all defined filters
+        for af in self.action_filters:
+            if af.component in context:
+                value = context[af.component]
+                # replace - with / in architecture component
+                if af.component == 'architecture':
```

Comment:
use constant instead of `'architecture'`

Reply:
done